enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,377 @@
+   already been updated by MCO, therefore it's ready for running OVN Kubernetes,
+   the sdn pod would just do "sleep infinity" rather than actually launching the
+   `openshift-sdn-node` process.
+
+3. CNO will also deploy OVN Kubernetes to the cluster with hybrid overlay
+   enabled. The ovnkube-node DaemonSet wrapper script also has the logic to
+   check if the bridge br-ex exists on the node. If `br-ex` doesn't exist, it
+   means the node has yet been updated by MCO, so instead of starting the
+   `ovnkube-node` process, it would run `hybrid-overlay-node` process that
+   discover the necessary hybrid overlay node information and annotate node
+   accordingly e.g.:
+
+   ```yaml
+   k8s.ovn.org/hybrid-overlay-distributed-router-gateway-mac: 00-c2-f5-92-28-ad
+   k8s.ovn.org/hybrid-overlay-node-subnet: 192.168.111.20/24
+   ```
+
+   otherwise it would launch the `ovnkube-node` process.
+
+#### Migration
+
+We can kick off the migration from CNO, by:
+
+```bash
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"spec":{"networkType":"OVNKubernetes"}'
+```
+
+CNO will update the `status.networkType` field of the `network.config` CR. It
+will trigger MCO to apply new MachineConfig to each node:
+
+1. MCO will 
+   - rerender the MachineConfig for each node
+   - try to cordon and drain the nodes based on
+     `MachineConfigPools.spec.maxUnavailable`
+2. CNO will be able to find a node is being drained by MCO by watching the MCO
+   node annotations, it will clear the `k8s.ovn.org/hybrid-overlay-xxx` node
+   annotation then set the annotation `k8s.ovn.org/node-subnets:` according to
+   the HostSubnet object of the node to bypass the ovnkube node subnet
+   allocation. 
+3. CNO will also update the hybrid overlay configuration and the onvkube-master
+   pods will take the updated hybrid overlay configuration then update the OVN
+   database.
+4. MCO will drain and reboot the node.
+5. After booting up, br-ex will be created and ovnkube-node will run in hybrid
+   overlay mode on the node.
+6. The wrapper script of ovnkube-node DaemonSet will see br-ex, then launch the
+   `ovnkube-node` process.
+7. MCO will uncordon the node. Pods will be created on the node using
+   ovn-kubernetes as the default CNI plugin.
+
+The above process will be repeated for each node, until all the nodes have been
+applied to the new MachineConfig and converted to OVN Kubernetes.
+
+#### Migration Cleanup
+
+Once migration is complete, CNO will:
+- delete the openshift-sdn DaemonSets
+- redeploy ovn-kubernetes in "normal" mode (no migration mode config, no node
+  affinity).
+- remove the migration-related labels from the nodes
+
+### API
+
+To start the migration, users need to update the `network.operator`
+CR by adding:
+
+CNO will deploy both OpenShift SDN and OVN Kubernetes pods in 'migration' mode.
+
+```json
+{ 
+  "spec": { 
+    "migration": {
+      "networkType": "OVNKubernetes",
+      "isLive": true
```

Comment:
what is the significance of "isLive"? Is this a new field with this enhancement?